### PR TITLE
[12.0][ENH] account_invoice_margin: add 'Margin(%)' field on Invoice analysis report

### PR DESCRIPTION
**Pull request by bodedra**
Taken screenshot from Runbot. http://3498077-102-f1859b.runbot1-2.odoo-community.org/web#action=273&model=account.invoice.report&view_type=pivot&menu_id=118

Screenshot 2021-08-13 at 11-33-15 Invoices Analysis - Odoo

### CODE REVIEW

**Comment by bodedra** [comment]
@pedrobaeza Would you please review this?

File: account_invoice_margin/report/account_invoice_report.py [hunk 1]
```diff
@@ -8,11 +8,13 @@ class AccountInvoiceReport(models.Model):
...
     def _select(self):
         select_str = super()._select()
-        return '%s, sub.margin' % select_str
+        return '%s, sub.margin, sub.margin_percent' % select_str
 
     def _sub_select(self):
         select_str = super()._sub_select()
-        return '%s, SUM(ail.margin_signed) AS margin' % select_str
+        return '%s, SUM(ail.margin_signed) AS margin, ' \
+               'ai.margin_percent AS margin_percent' % select_str
```
Comment (pedrobaeza) [inline account_invoice_margin/report/account_invoice_report.py]: This should have as well an aggregation formula (SUM, AVG...)

**Comment by bodedra** [comment]
@pedrobaeza Thanks for your suggestion and review. I have improved accordingly. Would you please review it again?

**Review by pedrobaeza — APPROVED** [review]

**Comment by bodedra** [comment]
@sergio-teruel are you available for review?

**Comment by pedrobaeza** [comment]
He's on holidays till September.

File: account_invoice_margin/report/account_invoice_report.py (hunk 1 above)
Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: ai.margin_percent or ail.ai.margin_percent will be best for avg?

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: ping @legalsylvain

File: account_invoice_margin/report/account_invoice_report.py [hunk 2]
```diff
@@ -8,11 +8,13 @@ class AccountInvoiceReport(models.Model):
...
     def _select(self):
         select_str = super()._select()
-        return '%s, sub.margin' % select_str
+        return '%s, sub.margin, sub.margin_percent' % select_str
 
     def _sub_select(self):
         select_str = super()._sub_select()
-        return '%s, SUM(ail.margin_signed) AS margin' % select_str
+        return '%s, SUM(ail.margin_signed) AS margin, ' \
+               'AVG(ai.margin_percent) AS margin_percent' % select_str
```
Comment (legalsylvain) [inline account_invoice_margin/report/account_invoice_report.py]: I think it's wrong for margin_percent. The average of margin is not the margin of whole invoices. To test, 
make a invoice with the same customer.
- an invoice of 1000€, margin 20% (margin 200€)
- a invoice of 2000€, margin 40%. (margin 800€)

when grouped by customer, with the current code, the margin will be 30%, that is wrong.

It is 200€ + 800€ = 1000 / 3000 = 33%.

You should recompute on the fly in my opinion.

**Review by legalsylvain — CHANGES_REQUESTED** [review]
the computation of _sub_select is wrong in my opinion.

otherwise, LGTM. 

thanks !

File: account_invoice_margin/report/account_invoice_report.py (hunk 2 above)
Comment (pedrobaeza) [reply account_invoice_margin/report/account_invoice_report.py]: Neither of both (sum or avg) has too much sense talking about percentage, but a group operator should be put. The only way to get the proper one is to inherit `read_group` and make the computation for each group, in a similar way I did this:

https://github.com/OCA/hr-attendance/blob/bcc7b04af14a52be994ebdc6ee9d52eab361303e/hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py#L239-L244

and it requires that the other measure is populated as well.

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: Thank you @legalsylvain for your example. I have changed percentage computation from invoice to invoice line.

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: @pedrobaeza awesome. Your example will help me a lot. Let me check something and work on more user test case.

**Comment by bodedra** [comment]
Thank you @Hardik-OSI for your contribution.

@legalsylvain @pedrobaeza 

I have tested according to your given example and validated results at runbot. http://3499288-102-2e7382.runbot3.odoo-community.org/web?debug=1#action=273&model=account.invoice.report&view_type=pivot&menu_id=118

**Invoice:**

Screenshot 2021-08-20 at 09-28-21 Invoices - Odoo

**Invoice Analysis report:**

Screenshot 2021-08-20 at 09-29-14 Invoices Analysis - Odoo

**Comment by bodedra** [comment]
@victoralmau are you available to review this PR?

**Review by legalsylvain — APPROVED** [review]
Thanks!

**Comment by pedrobaeza** [comment]
/ocabot merge major

Can you please fw-port it to upper versions?

**Comment by bodedra** [comment]
Yes @pedrobaeza I will do that.

**Pull request merged**